website/pages/blog/type-safe-your-api.mdx

```diff
@@ -0,0 +1,132 @@
+- The `topTracks` variable lacks type safety according to the API documentation.
+
+## How I Use Web API with feTS?
+To use the Spotify API with feTS, follow these steps:
+
+1. Create a new TypeScript project and install feTS from [here](https://the-guild.dev/openapi/fets/client/quick-start).
+
+2. Create a TypeScript file that exports your OpenAPI document as a constant:
+```typescript filename="openapi.ts"
+export default { openapi: '3.0.0' /* ... */ } as const;
+```
+In our case, we will use the Spotify API, so we can retrieve the OpenAPI document from [here](https://developer.spotify.com/reference/web-api/open-api-schema.yaml). Since this OpenAPI document is in YAML format, we need to convert it to JSON format before using it in our project. You can find an example of how to perform the conversion [here](https://github.com/TuvalSimha/feTS-example/blob/main/convert-yaml-to-json.ts).
+
+3. Now, we are ready to use feTS to create a type-safe API client:
+```typescript filename="spotify.ts"
+import { createClient, Mutable } from "fets";
+import spotifyOpenapi from "./spotifyOpenapi";
+
+const client = createClient<Mutable<typeof spotifyOpenapi>>({
+  endpoint: 'https://api.spotify.com/v1',
+});
+```
+
+We import the `createClient` and `Mutable` functions from the feTS package and use the `createClient` function to instantiate a new client. Furthermore, we import the `spotifyOpenapi` file we created in the previous step and use it as a generic type for the `createClient` function.
+
+4. Now, to retrieve our top 5 tracks from the last 30 days using the Spotify API, we can do the following:
+```typescript filename="spotify.ts"
+const topTracks = await client["/me/top/{type}"].get({
+  params: {
+    type: "tracks",
+  },
+  query: {
+    time_range: "short_term",
+    limit: 5,
+  },
+});
+```
+By using the client in this manner, we gain type safety for the endpoint, parameters, and query.
+
+![endpoint](/blog-assets/type-safe-your-api/endpoint.png)
+
+
+5. In conclusion, we know the type of the `topTracks` variable out-of-the-box.
+```typescript filename="spotify.ts"
+const topTracksResponse = await topTracks.json();
+console.log(topTracksResponse);
+```
+
+## Advantages and Disadvantages of Using feTS for Spotify API Integration
+
+### Advantages of feTS
+- Robust Type Safety: feTS eliminates the need for manual type definitions or code generation by inferring types from the OpenAPI specification. This ensures reliable and type-safe API interactions.
+- OpenAPI Compatibility: By leveraging the OpenAPI specification, feTS seamlessly integrates with various tools and frameworks, providing flexibility and convenience.
+- Developer-Friendly Experience: feTS simplifies the API consumption process, allowing developers to focus on writing clean and concise code.
+- No dependence on SDK maintainers: feTS allows you to use the API without waiting for the SDK maintainers to update the SDK.
```

Comment:
I would even say that now companies can give up on creating SDKs and they better just offer openapi and direct their users to feTS

Reply:
I would even mention that you had cases where the provider API has an SDK and you still chose to use feTS